logstash-core/lib/logstash/webserver.rb

```diff
@@ -189,39 +192,43 @@ def _init_server
       ::Puma::Server.new(@app, events)
     end
 
-    def bind_to_available_port
-      http_ports.each_with_index do |candidate_port, idx|
-        begin
-          break unless running?
-
-          @server = _init_server
-
-          logger.debug("Trying to start API WebServer", :port => candidate_port, :ssl_enabled => ssl_enabled?)
-          if @ssl_params
-            unwrapped_ssl_params = {
-              'keystore'      => @ssl_params.fetch(:keystore_path),
-              'keystore-pass' => @ssl_params.fetch(:keystore_password).value
-            }
-            ssl_context = Puma::MiniSSL::ContextBuilder.new(unwrapped_ssl_params, @server.events).context
-            @server.add_ssl_listener(http_host, candidate_port, ssl_context)
-          else
-            @server.add_tcp_listener(http_host, candidate_port)
-          end
-
-          @port = candidate_port
-          logger.info("Successfully started Logstash API endpoint", :port => candidate_port, :ssl_enabled => ssl_enabled?)
-          set_http_address_metric("#{http_host}:#{candidate_port}")
-
-          @server.run.join
-          break
-        rescue Errno::EADDRINUSE
-          if http_ports.count == 1
-            raise Errno::EADDRINUSE.new(I18n.t("logstash.web_api.cant_bind_to_port", :port => http_ports.first))
-          elsif idx == http_ports.count-1
-            raise Errno::EADDRINUSE.new(I18n.t("logstash.web_api.cant_bind_to_port_in_range", :http_ports => http_ports))
+    def create_server_thread
+      server_thread = nil
+      @mutex.synchronize do
+        @server = _init_server
```

Comment:
`@server` is no longer discarded, as before, between `bind_to_port` attempts.
:white_check_mark: looking at [`add_tcp_listener`](https://github.com/puma/puma/blob/v5.6.3/lib/puma/binder.rb#L347=) and [`add_ssl_listener`](https://github.com/puma/puma/blob/v5.6.3/lib/puma/binder.rb#L380=) impl 
there are no side-effects before `tcp_server.listen` happens ...